static/js/narrow.js

```diff
@@ -643,6 +644,7 @@ exports.to_compose_target = function () {
 
 function handle_post_narrow_deactivate_processes() {
     compose_fade.update_message_list();
+    notifications.hide_history_limit_message();
```

Comment:
This should be `exports.hide_or_show_history_limit_message(home_msg_list)`, I think.  Part of fixing the bug that you currently never show the notice for the home_msg_list.